Switch to CSS variables

Squidly271 opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Improved mobile menu controls and sidebar interactions for easier navigation.
  * Added alternate view and menu overlay for enhanced UI flexibility.
  * Introduced new CSS theme options (Azure, Black, Gray, White) and centralized color theming.
  * Added plugin download script for streamlined plugin management.

* **Enhancements**
  * Refined UI structure and styling for consistency and better mobile support.
  * Updated terminology to "Boot Device" in messages for clarity.
  * Improved error handling and long-loading feedback in the interface.
  * Enhanced keyboard navigation and accessibility.

* **Bug Fixes**
  * Corrected UI inconsistencies and improved popup/modal behaviors.
  * Fixed issues with event handling and menu state restoration.

* **Style**
  * Major CSS refactor: modularized styles, standardized button and card appearances, and improved responsive layouts.

* **Chores**
  * Updated internal resource paths and checksums to reflect new and reorganized files.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Squidly271 committed:
Switch to CSS variables

Squidly271 committed:
Separate render routines to separate functions per app type

Squidly271 commented:
Skin.php needs to be in branch as further refinements to CSS will require code updates and major merge conflict will result due to the big refactor on skin.php

Squidly271 commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin.php:
```diff
@@ -1714,7 +1131,864 @@ function displayPopup($template) {
+      $actionsContext[] = ["icon"=>"ca_fa-switchto","text"=>$template['SwitchLanguage'],"action"=>"CAswitchLanguage('$countryCode');"];
+    }
+    } else {
+    $actionsContext[] = ["icon"=>"ca_fa-install","text"=>$template['InstallLanguage'],"action"=>"installLanguage('{$template['TemplateURL']}','$countryCode');"];
+  }
```
TemplateURL is generated by an associated script and is not set by a user.  This is safe, and adding slashes may introduce other issues

Squidly271 commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin.php:
```diff
@@ -1714,7 +1131,864 @@ function displayPopup($template) {
+      // ... existing code from displayPopup for repository ...
+      // (We will leave this as-is for now)
+  }
+}
```
Blame your cursor AI counterpart.  This is a draft PR.  But, there are no "unknown" app types

Squidly271 commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin.php:
```diff
@@ -1714,7 +1131,864 @@ function displayPopup($template) {
+        if ((($template['RequiresFile'] && is_file($template['RequiresFile'])) || ! $template['RequiresFile'])) {
+          $buttonTitle = $template['InstallPath'] ? tr("Reinstall") : tr("Install");
+          $isDeprecated = $template['Deprecated'] ? "&deprecated" : "";
+          $isDeprecated = $template['Compatible'] ? "&incompatible" : $isDeprecated;
```
I am aware.  But more coding changes throughout CA is required to handle the inverse.

Squidly271 committed:
Revert "Separate render routines to separate functions per app type"

This reverts commit bb5cbd1d63f07a8907230ea1685c60ac199dc008.

Squidly271 committed:
Cleanup CSS

Squidly271 committed:
Fixed: Clicking dockerhub from a dockerhub search would bypass the "external link" popup

Squidly271 committed:
CSS Cleanup

Squidly271 committed:
On official containers, display "Official container" instead of the repository name

Came about via a Techno Tim video where confusion resulted as to who wrote Official Jellyfin

Squidly271 committed:
Restore missing mfp styles

Squidly271 committed:
When only a single app is displaying on home screen, change it to two apps to display vertically

Squidly271 committed:
Hide menu when card won't fit and account for themes and font size dicrepancies

Squidly271 committed:
publish beta .plg

Squidly271 committed:
check for updates on beta

Squidly271 committed:
Handle errors caused by browser back / forward in certain circumstances

Squidly271 committed:
Select branch fontsize too large

Squidly271 committed:
Fix settings page wide buttons

Squidly271 committed:
Override footer to remain at bottom in low res

Squidly271 committed:
Fix menu bar not remaining in place

Squidly271 committed:
Update community.applications-css.plg

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/Apps.page:
```diff
@@ -145,22 +145,71 @@ function tr($string,$ret=false) {
-<style>
+$font = $display['font'] ?: "62.5";
+$themeOffset = ($display['theme'] == "azure" || $display['theme'] == "gray") ? $font + 10 : 0;
 
-<?
- include ("/$docroot/plugins/$plugin/skins/Narrow/css.php");
+switch ($font) {
+  case "50":
+    $doubleRow = 827;
+    $noMenu = 500;
+    break;
+  case "56.25":
+    $doubleRow = 922;
+    $noMenu = 558;
+    break;
+  case "62.5":
+    $doubleRow = 1016;
+    $noMenu = 608;
+    break;
+  case "68.75":
+    $doubleRow = 1111;
+    $noMenu = 653;
+    break;
+  case "75":
+    $doubleRow = 1205;
+    $noMenu = 719;
+    break;
+  case "80":
+    $doubleRow = 1281;
+    $noMenu = 763;
+    break;
+    
+}
+$doubleRow += $themeOffset;
+$noMenu += $themeOffset;
```
This isn't ideal to have to calculate this based on font size but I know that with the way CA is configured to display results that you're using CSS to limit the shown results.

Ideally you wouldn't need these "magic" numbers. And instead your PHP determines the results that are shown. And then your CSS is setup to flow naturally regardless of the height of things.

But if this is working with how CA is setup today then it's fine.

Squidly271 replied:
Trust me - I tried to do this in code.  Problem is that in the GUI there's still a mixture of rem and px for sizing, and there was no formula that would come up with the exact number I wanted.  Reason its done to the pixel and for each font size was simply because I needed a media inquiry to switch the home page from one row to two rows, but if the px was off by 1 or 2 then I'd get two rows with two columns.   My OCD kicked in on that.   Wasn't a fun time sizing a browser to get the correct value for each theme and for each font-size.  (And if I made any size changes on elements afterwards I had to start over again)

Well aware that I'll likely have to do this again, but do have a thought on how to get JS to figure it out and send the appropriate number of apps to the backend for the home page...

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/ca_settings.page:
```diff
@@ -143,7 +143,7 @@ _(Enable developer mode)_:
 
 &nbsp;
-: <input id='debugging' type='button' onclick='debug();' value='DOWNLOAD LOG' style='display:none;'></input><input type="submit" name="#apply" value="<?tr("Apply");?>" id='Apply'><input type="button" id="DONE" value="<?tr("Done");?>" onclick="done()">
+: <span><input id='debugging' type='button' onclick='debug();' value='DOWNLOAD LOG' style='display:none;'></input></span><span><input type="submit" name="#apply" value="<?tr("Apply");?>" id='Apply'><input type="button" id="DONE" value="<?tr("Done");?>" onclick="done()"></span>
```
A single `<span>` to wrap both buttons would likely be fine. Instead of individually wrapping each button.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/include/helpers.php:
```diff
@@ -653,9 +653,9 @@ function formatTags($leadTemplate,$rename="false") {
 
     $o = "<table>";
-    $o .= "<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td><a class='appIconsPopUp xmlInstall ca_normal' data-type='$type' data-xml='{$template['Path']}'>Default</a></td><td class='appIconsPopUp xmlInstall ca_normal' data-type='default' data-xml='{$template['Path']}'>".tr("Install Using The Template's Default Tag")." (<span class='ca_bold'>:$defaultTag</span>)</td></tr>";
+    $o .= "<tr><td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</td><td><a class='xmlInstall ca_normal' data-type='$type' data-xml='{$template['Path']}'>Default</a></td><td class='xmlInstall ca_normal' data-type='default' data-xml='{$template['Path']}'>".tr("Install Using The Template's Default Tag")." (<span class='ca_bold'>:$defaultTag</span>)</td></tr>";
```
Using `&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;` for spacing is naughty and typically a no no. But I realize this isn't new per se so don't change this right now.

Squidly271 replied:
I know.  That sub section isn't high on my todo list

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin.html:
```diff
@@ -1,79 +1,77 @@
+      <li class='caMenuItem sectionMenu caPinnedMenu onlyShowWithFeed' data-category='pinned_apps'><?tr("Pinned Apps");?></li>
+      <li class='caMenuItem favouriteRepo onlyShowWithFeed'><?tr("Favourite Repo");?></li>
+      <li class='caMenuItem sectionMenu actionCentre onlyShowWithFeed' style='display:none;' data-category='action_centre'><?tr("Action Centre");?></li>
+      <hr class='onlyShowWithFeed'>
```
Nice - we love code formatting :)

There's a mixture of things happening in the CSS class nomenclature. Some things are prefixed with `ca` and some things aren't.

I would consider standardizing this in the future. Definitely not any deal breaker right now.

Squidly271 replied:
All depended upon my mood when originally doing it...   The prefixing of vars with "ca" would be a higher priority

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin.html:
```diff
@@ -1,79 +1,77 @@
+      <hr class='category_hr'>
+      <li class='categoryMenu caMenuItem caRepositoryMenu onlyShowWithFeed nonDockerSearch' data-category='repos'><?tr("Repositories");?></li>
+    <hr class='category_hr onlyShowWithFeed'>
+      <li class='caMenuItem' onclick='window.location="/Apps/ca_settings"'><?tr("Settings");?></a></li>
```
Good suggestion

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin.html:
```diff
@@ -1,79 +1,77 @@
-  <hr class='onlyShowWithFeed'>
+<div class='ca_display_area'>
+  <div class='menuItems'>
+    <ul class='caMenu '>
+      <li class='startupButton caMenuItem onlyShowWithFeed' data-category='new'><?tr("Home");?></li>
+      <hr class='onlyShowWithFeed'>
+      <li class='caMenuItem sectionMenu onlyShowWithFeed' data-category='installed_apps'><?tr("Installed Apps");?></li>
+      <ul class='subCategory onlyShowWithFeed'>
+        <li class='caMenuItem sectionMenu onlyShowWithFeed dockerSpecific' data-category='inst_docker'><?tr("Docker");?></li>
+        <li class='caMenuItem sectionMenu onlyShowWithFeed' data-category='inst_plugins'><?tr("Plugins");?></li>
+      </ul>
+      <li class='caMenuItem sectionMenu onlyShowWithFeed' data-category='previous_apps'><?tr("Previous Apps");?></li>
+      <ul class='subCategory onlyShowWithFeed'>
+        <li class='caMenuItem sectionMenu dockerSpecific' data-category='prev_docker'><?tr("Docker");?></li>
+        <li class='caMenuItem sectionMenu' data-category='prev_plugins'><?tr("Plugins");?></li>
+      </ul>
+      <li class='caMenuItem sectionMenu caPinnedMenu onlyShowWithFeed' data-category='pinned_apps'><?tr("Pinned Apps");?></li>
+      <li class='caMenuItem favouriteRepo onlyShowWithFeed'><?tr("Favourite Repo");?></li>
+      <li class='caMenuItem sectionMenu actionCentre onlyShowWithFeed' style='display:none;' data-category='action_centre'><?tr("Action Centre");?></li>
+      <hr class='onlyShowWithFeed'>
 
```
good suggestion if it doesn't break anything in the webgui.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin.html:
```diff
@@ -1,79 +1,77 @@
-  <hr class='onlyShowWithFeed'>
+<div class='ca_display_area'>
+  <div class='menuItems'>
+    <ul class='caMenu '>
```
rouge space after `caMenu` that could be removed

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin.html:
```diff
@@ -1,79 +1,77 @@
+      <li><font size='0'>Modified</font></li>
+    <?endif;?>
+    <hr>
+      <li class='caMenuItem debugging'><?tr("Debugging");?></li>
```
spacing alignment is a bit funky here

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin.html:
```diff
@@ -1,79 +1,77 @@
+  <div class='mainArea'>
+  <div class='spinnerBackground'></div>
+  <div class='spinner fixed'></div>
+  <strong><font size='3'></strong></font>
```
great suggestion. `<font>` tags are a thing of the past

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/font

additionally the ordering of these tags are incorrect anyway.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin.html:
```diff
@@ -1,79 +1,77 @@
+  <div class='spinnerBackground'></div>
+  <div class='spinner fixed'></div>
+  <strong><font size='3'></strong></font>
+  <div class='searchArea onlyShowWithFeed'><span id='searchFilter'><input type='text' id='searchBox' placeholder='<?tr('Search');?>'><span class='searchSubmit' onclick='doSearch(true)'><i id='searchButton' class='fa fa-search fa-lg'></i></span><span class='caButton maxPerPage' id='maxPerPage' style='display:none;'></span></div>
```
good callout - formatting these lines will make finding missing tags much easier

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin.html:
```diff
@@ -1,79 +1,77 @@
+  <div class='mainArea'>
+  <div class='spinnerBackground'></div>
+  <div class='spinner fixed'></div>
+  <strong><font size='3'></strong></font>
+  <div class='searchArea onlyShowWithFeed'><span id='searchFilter'><input type='text' id='searchBox' placeholder='<?tr('Search');?>'><span class='searchSubmit' onclick='doSearch(true)'><i id='searchButton' class='fa fa-search fa-lg'></i></span><span class='caButton maxPerPage' id='maxPerPage' style='display:none;'></span></div>
+  <div id="Category" class="categoryLine">&nbsp;</div>
+  <div class='dockerSearch' style='display:none'><a class='initDockerSearch'><?tr("Click Here To Get More Results From DockerHub");?></a></div>
+  <div id='sortIconArea' style='display:none;'><?tr("Sort By:");?> <a id='defaultSort' class='sortIcons enabledIcon' data-sortBy='Name' data-sortDir='Up'><?tr("Name Ascending");?></a><a class='sortIcons' data-sortBy='Name' data-sortDir='Down'><?tr("Name Descending");?></a><a class='sortIcons' data-sortBy='downloads' data-sortDir='Down'><?tr("Downloads");?></a><a class='sortIcons' data-sortBy='Fir ...
+  <div class='multi_installDiv'><input class='multi_install multi_installButton' type='button' value='<?tr("Install Selected Applications");?>' disabled><input type='button' class='multi_installAll' value='<?tr("Select all on page");?>'><input type='button' class='multi_installClear' value='<?tr("Clear");?>' disabled><input type='button' class='multi_deleteButton' style='visibility:hidden;' value ...
+  <div id='templates_content'><div class='ca_templatesDisplay'><font size='6'><?tr("Updating Content");?></font></div></div>
+  <div class='multi_installDiv'><input class='multi_install multi_installButton' type='button' value='<?tr("Install Selected Applications");?>' disabled><input type='button' class='multi_installAll' value='<?tr("Select all on page");?>'><input type='button' class='multi_installClear' value='<?tr("Clear");?>' disabled><input type='button' class='multi_deleteButton' style='visibility:hidden;' value ...
+  <span id='debugging'></span>
+  <span id='caCredits' style='display:none;'>
+    <div class='credits'>
+      <div class='ca_center'><i class='fa fa-users' style='font-size:5rem;margin-top:2rem;'></i></div>
+      <div class='ca_center'><font size='5'>Community Applications</font></div>
+      <div class='ca_center ca_creditheader'><?tr("Development");?></div>
```
great call out

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin.html:
```diff
@@ -1,79 +1,77 @@
+  <div id='sortIconArea' style='display:none;'><?tr("Sort By:");?> <a id='defaultSort' class='sortIcons enabledIcon' data-sortBy='Name' data-sortDir='Up'><?tr("Name Ascending");?></a><a class='sortIcons' data-sortBy='Name' data-sortDir='Down'><?tr("Name Descending");?></a><a class='sortIcons' data-sortBy='downloads' data-sortDir='Down'><?tr("Downloads");?></a><a class='sortIcons' data-sortBy='Fir ...
+  <div class='multi_installDiv'><input class='multi_install multi_installButton' type='button' value='<?tr("Install Selected Applications");?>' disabled><input type='button' class='multi_installAll' value='<?tr("Select all on page");?>'><input type='button' class='multi_installClear' value='<?tr("Clear");?>' disabled><input type='button' class='multi_deleteButton' style='visibility:hidden;' value ...
+  <div id='templates_content'><div class='ca_templatesDisplay'><font size='6'><?tr("Updating Content");?></font></div></div>
+  <div class='multi_installDiv'><input class='multi_install multi_installButton' type='button' value='<?tr("Install Selected Applications");?>' disabled><input type='button' class='multi_installAll' value='<?tr("Select all on page");?>'><input type='button' class='multi_installClear' value='<?tr("Clear");?>' disabled><input type='button' class='multi_deleteButton' style='visibility:hidden;' value ...
```
consider some formatting here and making this more readable at a quick glace. HTML elements on the same line can be hard to quickly parse through are is prone to errors.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/Apps.page:
```diff
@@ -145,22 +145,71 @@ function tr($string,$ret=false) {
+
+
+
+<link rel='stylesheet' href='<?autov("/plugins/community.applications/skins/Narrow/themes/theme-$theme.css");?>'>
```
Could potentially simplify `themes/theme-$theme.css` into `themes/$theme.css`

That would match the individual CSS var files I did for the webgui.

Squidly271 replied:
Somewhat surprised that you didn't question /skins/Narrow/themes/theme-$theme.css

"Narrow" is a hold-over because at one point there were actually 3 different skins to choose from:

Table / list view (didn't last long)
Narrow (Basically icon only)
Wide (What we see now)

Jon insisted that Narrow was what everyone would want to see (I disagreed), so it was the default.  Since everyone hated it, I dropped it and went with strictly wide.  Naming it Narrow was simply because of how the default was on previous CA versions.

But I do still support 3rd party CSS overrides if someone ever chooses to take advantage of that (can't remember if Theme Engine does create the applicable file(s) for CA

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/themes/theme-azure.css:
For each of your theme files consider tapping in to variables that already exist and are set by the webgui. That way we're eliminating any potential one-off color usage and being more consistent with the built in themes.

Take a look @ https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

This are my "standardized" colors and are used throughout each of the webgui's theme files.

See if you're able to use any values that are already set here via the webgui.

You should also take a look @ https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/themes/azure.css and the other webgui theme files to see if there's values set in each of the themes that you could tap in to rather then setting your own hex colors.

When I was consolidating all the hex colors in the webgui, I would often find very similar hex colors and just pick one to be the "main" one for the shade of color. I would do this by using some hex code viewer tool and pasting similar ones in and just deciding what a color might be similar to in the already existing color palette.

So in the webgui's `default-color-palette.css` you'll see comments like `/* Condensed from: <hex_code_x> <hex_code_y> etc ... */`. The ones in the comments are no longer used are the var they're commented with becomes the top dog for that shade.

Let me know if you have any questions here.

Squidly271 replied:
Its in progress. Removing unused colours from the CSS, and using the OS defined with fall backs for < 7.2

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+/* Base Styles */
+body {
+    scrollbar-gutter: stable;
```
Starting from this line and going to line 1912 your spacing is bumped in a one step too much. Fix the formatting so the top level rules align

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    top: 2rem;
+    display: block;
+    overflow-x: hidden;
+    min-height: 90vh;
```
This min-height might be giving you some trouble. Ideally shouldn't be needed.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .menuItems {
+    position: relative;
+    top: 2rem;
```
Using `top`, `right`, `bottom`, or `left` with `position: relative` is going to give you some headaches and throw off the flow of things where you will have to unexpectedly over compensate for other in elements.

Try using `margin-top` instead.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    height: auto;
+    align-items: stretch;
+    min-width: 14rem;
+    max-width: 14rem;
```
Just set a `width` instead of both these.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    font-size: 1.4rem !important;
+  }
+  
+  .ca_note {
+    color: #d67777;
+  }
+  
+  .ca_red {
+    color: #882626;
+  }
+  
+  .ca_green {
+    color: #038503;
+  }
```
trying using values from the [webgui's default-color-palette.css](https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css)

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  body.stop-scrolling {
+    height: 70%;
```
Avoid setting a fixed percentage height on the body like this

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    float: left;
+    height: 20rem;
+    min-width: 24rem;
+    max-width: 24rem;
```
try just using `width` instead of both rules here

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    /* Layout */
+    display: block;
+    position: relative;
+    top: 0.5rem;
```
use `margin-top` instead of `top` for position relative elements.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  .caButton {
+    /* Layout */
+    display: inline-block;
+    margin-right: 1rem;
```
ideally you wouldn't set a margin on a button. but instead have a parent element around buttons that determines the spacing between them.

that said, i realize the webgui has a margin on buttons. Which in the near future I hope to change.

so don't feel like you need to change anything here.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .caButton:hover {
+    background: var(--unraid-66-color);
+    color: #fff !important;
```
use a var from https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .caButton:hover a,
+  a.caButton:hover {
+    color: #fff;
```
use a var from https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  .selectedMenu {
+    color: var(--unraid-66-color);
```
a more descriptive color would be good

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    transition: 0.5s;
+  }
+  
+  .sidenavHide {
+    width: 0;
+  }
+  
+  .sidenavShow {
+    width: 90%;
+    max-width: 70rem;
+  }
```
CSS transitions for `width` a very taxing and prone to frame rate lag.

Consider keeping the width consistent and instead use `transform: translate()` and opacity to determine visibility of this element.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  /* Animations */
+  @keyframes cycleRed {
+    0% { color: inherit; }
+    50% { color: #882626; }
```
Use an existing var here https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    position: relative;
+    
+    /* Typography */
+    color: red;
```
Use an existing var here https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    padding: 0;
+    
+    /* Visual */
+    background: linear-gradient(to bottom right, white, hsla(0, 0%, 100%));
```
Use an existing var here for the colors if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    
+    /* Visual */
+    background: linear-gradient(to bottom right, white, hsla(0, 0%, 100%));
+    border: 1px solid rgba(0, 0, 0, 0.3);
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    background: linear-gradient(to bottom right, white, hsla(0, 0%, 100%));
+    border: 1px solid rgba(0, 0, 0, 0.3);
+    border-radius: 0.3em;
+    box-shadow: 0.05em 0.2em 0.6em rgba(0, 0, 0, 0.2);
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  .awesomplete li[aria-selected="true"] mark {
+    background: hsl(86, 100%, 21%);
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    width: 9rem;
+    position: relative;
+    left: -10rem;
+    margin-right: -9rem;
```
position relative and top, right, bottom, left can lead to trouble.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  .caChart {
+    height: 1px;
+    width: 3px;
```
This is definitely a little strange

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+
+  .ca_author {
+    font-size: 1rem;
+    margin-left: 0.75rem;
+    margin-top: 0.75rem;
```
Nitpick - in the future let's try to get something to automatically organize your rules for you.

It's often best to follow the natural flow of CSS and organize rules similar to how the native `margin` / `padding` rule would - TRBL aka top, right, bottom, left.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  .ca_fa-checkmark {
+    color: #038503;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  .ca_fa-delete {
+    color: #882626;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  .ca_private {
+    color: #882626;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .ca_readmore:hover {
+    text-decoration: none;
+    color: #d67777;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  .ca_serverWarning {
+    color: #cecc31;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  .ca_stat {
+    color: coral;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    font-size: 1.5rem;
+  }
+  
+  .ca_template {
+    color: #606E7F;
+    border-radius: 0px 0px 2rem 2rem;
+    display: inline-block;
+    text-align: left;
+    overflow: auto;
+    height: 27rem;
+    width: 36rem;
+    padding-left: .5rem;
+    padding-right: .5rem;
+    background-color: #DDDADF;
+  }
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  .ca_template {
+    color: #606E7F;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    width: 36rem;
+    padding-left: .5rem;
+    padding-right: .5rem;
+    background-color: #DDDADF;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  .ca_template_icon {
+    color: #606E7F;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    width: 37rem;
+    float: left;
+    display: inline-block;
+    background-color: #C7C5CB;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+  }
```
love seeing this...love automatic spacing continuity

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .dockerHubStar::before {
+    content: "\e95a";
+    font-family: UnRaid;
```
Make sure we're using the right case here. First time seeing the R capitalized in the font-family rule.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .favCardBackground {
+    float: right;
+    color: #bb0000;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .greenCardBackground {
+    clip-path: polygon(0 0,100% 0, 100% 100%);
+    background-color: #008100;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    width: 9rem;
+    position: relative;
+    left: -10rem;
+    margin-right: -9rem;
```
position relative with a mix of top, left, and margin-right might be causing a headache

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  .homeMore:hover {
+    color: #d67777;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .homespotlightIconArea {
+    display: inline-block;
+    position: relative;
+    top: -25rem;
+    left: 24rem;
```
another position relative with top and left to be careful of

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .installedCardBackground {
+    clip-path: polygon(0 0,100% 0, 100% 100%);
+    background-color: #322fff;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .installedPopup {
+    clip-path: polygon(0 0,100% 0, 100% 100%);
+    background-color: #322fff;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .officialCardBackground {
+    clip-path: polygon(0 0,100% 0, 100% 100%);
+    background-color: #854153;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .LTOfficialCardBackground {
+    clip-path: polygon(0 0,100% 0, 100% 100%);
+    background-color: #F15A2C;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .pinnedCard {
+    float: right;
+    color: #1fa67a;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .popUpBack {
+    font-size: 1.5rem;
+    color: #f34646;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  .popUpClose {
+    margin-top: 2rem;
+    font-size: 1.5rem;
+    color: #f34646;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  i.popupIcon {
+    color: #626868;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  }
+  
+  .spotlightMessage a:hover {
+    color: #d67777;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  
+  .warningCardBackground {
+    clip-path: polygon(0 0,100% 0, 100% 100%);
+    background-color: #810000;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    width: 9rem;
+    position: relative;
+    left: -10rem;
+    margin-right: -9rem;
```
another position relative with top and left to be wary of

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+
+  @keyframes cycleRed {
+    0% { color: inherit; }
+    50% { color: #882626; }
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  z-index: 1042;
+  overflow: hidden;
+  position: fixed;
+  background: #0b0b0b;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+}
+
+.mfp-preloader {
+  color: #CCC;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+}
+
+.mfp-preloader a {
+  color: #CCC;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+}
+
+.mfp-preloader a:hover {
+  color: #FFF;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+}
+
+.mfp-close-btn-in .mfp-close {
+  color: #333;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+  position: absolute;
+  top: 0;
+  right: 0;
+  color: #CCC;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+}
+
+.mfp-arrow-left:after {
+  border-right: 17px solid #FFF;
+  margin-left: 31px;
+}
+
+.mfp-arrow-left:before {
+  margin-left: 25px;
+  border-right: 27px solid #3F3F3F;
+}
+
+.mfp-arrow-right {
+  right: 0;
+}
+
+.mfp-arrow-right:after {
+  border-left: 17px solid #FFF;
+  margin-left: 39px;
+}
+
+.mfp-arrow-right:before {
+  border-left: 27px solid #3F3F3F;
+}
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+}
+
+.mfp-figure small {
+  color: #BDBDBD;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+.mfp-title {
+  text-align: left;
+  line-height: 18px;
+  color: #F3F3F3;
```
Use an existing var here if possible https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
If you need to, define CA specific CSS vars for colors that are CA specific that might be re-used throughout this file. That said, I heavily implore using existing CSS vars provided by the webgui where possible.

zackspear replied:
Another thing, I realize that a lot of this already existed.

So don't feel pressured to have to get things changed quickly.

There's a lot of things I'm seeing here where I can clearly see you fighting against existing webgui CSS and even your own CSS.

I think there's a possibility in the future that we can heavily reduce the amount of CSS here and simplify things. But in due time :)

Squidly271 replied:
WIP.  There was a fair amount defined that hadn't been removed when it wasn't in use anymore, and combining multiple classes that are either identical or only slightly different into a single definition.

But, bear in mind that when I started this project, I didn't know anything about CSS.  Implementation of Flex Boxes for the display was when everything moved to CSS (mostly), and I'm the first to admit that certain ways I accomplished stuff on placement wasn't the proper way to do things, but it did work.  

Part of the issue on things is that the basic framework of the back end hasn't changed over the years, and because of that framework, as new features have come along they've had to be forced in to that framework.

If CA didn't actually work, then a complete rewrite would be high on the list.  That rewrite though is now being slowly phased in (particularly with the actual skin code) which initially makes new app types easier to add in, and will ultimately allow for a transition to client skin rendering, followed by more transitions from the server to the client of the backend.  It sucks having a project that is so prevalent in the user base while also being bug free (except for corner cases that only I see that can't be fixed easily in the code) and having to maintain backwards compatibility.

> fighting against GUI

Still not a fan of having to do an insertAfter to move an element (mainly the Search Area) to be sticky as you scroll, but no way around that unfortunately that I could find.   Luckily, your DOM between sidenav and top nav are very close, so no trickery to get what worked on top nav to work on side nav.   Older OS versions, on top nav just needed to pick a different element to insert after, but for side nav it was a major pain so just dropped that feature on legacy OS.

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/themes/theme-black.css:
Look @ https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css and https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/themes/black.css for existing Black theme vars

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/themes/theme-gray.css:
Look @ https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css and https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/themes/gray.css for existing gray theme vars

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/themes/theme-white.css:
Look @ https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/default-color-palette.css and https://github.com/unraid/webgui/blob/3cfa60e726b4a9df8ab581dcc1174f966211d114/emhttp/plugins/dynamix/styles/themes/white.css for existing white theme vars

zackspear reviewed (changes requested):
There's a good number of places we can consolidate one off color usage

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    height: 2rem;
+  }
+
+  .ca_display_area {
+    display: flex;
+  }
```
```css
.caContentWrapper {
  display: grid;
  grid-template-columns: 1fr;
  gap: 4rem;
  width: 100%;
  padding: 1rem;
  box-sizing: border-box;
}

/* make this media query max-width what you think works best */
@media (max-width: 768px) {
  .caContentWrapper {
    grid-template-columns: 14rem 1fr;
  }
}
```

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    display: flex;
+  }
+
+  .mainArea {
+    right: 0px;
+    top: 2rem;
+    display: block;
+    overflow-x: hidden;
+    min-height: 90vh;
+    padding-left: 4.5rem;
+    align-items: stretch;
+  }
```
If you use the grid method I mention here - https://github.com/unraid/community.applications/pull/4/files#r2141266907

You could get rid of most of these rules and simplify down to

```css
.mainArea {
    display: block;
    overflow-x: hidden;
}
```

zackspear commented on source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/styles/community.applications.css:
```diff
@@ -0,0 +1,2392 @@
+    align-items: stretch;
+    min-width: 14rem;
+    max-width: 14rem;
+  }
```
If you use the grid method suggested for `.ca_display_area` you don't need any of these rules for `.menuItems` as they'd be redundant

Squidly271 committed:
Reorganization

Squidly271 committed:
Switch to modal-like for stats and credits

Squidly271 committed:
@media value tweaking, hide maxperpage when single column

Squidly271 committed:
If searchbox has focus, don't open sidebar on hover

Squidly271 committed:
allow arrow keys to change page.  Any key exits sidebar or alt display

Squidly271 committed:
overlay refinements, similar docker searches

Squidly271 committed:
DockerSearch buttons refinement

Squidly271 committed:
Only allow ESC to close sidebar/alternative view and disallow page changes

Squidly271 committed:
Fixed: Single option support links were opening twice

Squidly271 committed:
don't show docker search on repo searches

Squidly271 committed:
typo

Squidly271 committed:
coderabbit suggestions

Squidly271 committed:
Beta release

Squidly271 committed:
coderabbit fixes

Squidly271 committed:
...

Squidly271 commented:
@zackspear All colours are now vars.   Using OS vars will eventually come but need to maintain backwards compatibility, so this is my starting point.

Squidly271 committed:
if proxy failure, try without proxy

Squidly271 committed:
below

Add hamburger.
Reposition dockerSearch & templateSearch buttons
Fix settings field size
Simpler post functions
Styling on close / back buttons on sidebar now match button styles

Squidly271 committed:
Fix nesting

Squidly271 committed:
Handle extra scroll bar now appearing

Squidly271 committed:
various

Squidly271 committed:
below

Move some translations from javascript to php
Change timeouts on downloads
Add popup if post calls taking longer than expected.
Ability to abort post if its taking an extreme long time
Close curl channel after downloads or when trying again if proxy doesn't appear to work

Squidly271 committed:
release

Squidly271 committed:
below

Change wording of Flash Drive to boot device
Refine mobile menu
Don't show max per page if results are <24
Search Area now always visible
Page navigation now always visible
Use swal modal during initial updating content

Squidly271 committed:
Below

Wrap more error checking on JSON.parse
7.1.4 fixes

Squidly271 committed:
Colour tweaks

Squidly271 commented on source/community.applications/usr/local/emhttp/plugins/community.applications/Apps.page:
```diff
@@ -342,21 +484,30 @@ $(function(){
   }
 
-  caPluginUpdateCheck("community.applications.plg",{
+  addBannerWarning("DO NOT RELEASE AS IS - update check requires filename change");
+  caPluginUpdateCheck("community.applications-css.plg",{
```
I'm very well aware of this.  Hence why I've got the banner to remind me.

Squidly271 replied:
I'm also aware of the magic numbers.  Not ideal situation, but only solution at this moment for the problem

Squidly271 committed:
CSS Consolodation

Squidly271 commented on source/community.applications/usr/local/emhttp/plugins/community.applications/javascript/helpers.js:
```diff
@@ -151,78 +131,70 @@ function post(options,callback) {
+  
+    
+  $.post(execURL,options).done(function(result) {
+    if (result.script) {
+      try {
+        eval(result.script);
+      } catch(e) {
+        alert("Could not execute result.script "+e);
+      }
+    }
+    if (result.globalScript) {
+      try {
+        eval(result.globalScript);
+      } catch(e) {
+        alert("Could not execute result.globalScript "+e);
+      }
+    }
```
No real workable solution at this point in time.   The eval code is strictly server based, doesn't contain any user defined input.  Not an issue, but agreed its not ideal

Squidly271 committed:
CSS Refinements

Squidly271 committed:
Simplifications

Squidly271 committed:
Update community.applications-2025.07.14-x86_64-1.txz

Squidly271 committed:
Coderabbit catches

Squidly271 committed:
Update skin.html

Squidly271 commented on source/community.applications/usr/local/emhttp/plugins/community.applications/scripts/downloadPlugin:
```diff
@@ -0,0 +1,72 @@
+write("Downloading plugin\n");
+
+$url = $argv[3];
+$result = download_url($url,"/tmp/community.applications/".basename($url));
+if ( $result === false ) {
+    write("Download failed\n");
+    @unlink("/tmp/community.applications/".basename($url));
+    done(0);
+} else {
+    run("plugin install /tmp/community.applications/".basename($url));
+    write("Plugin installed\n");
+    @unlink("/tmp/community.applications/".basename($url));
+    done(0);
+}
```
Not an issue.  The plugin command does all of that

Squidly271 committed:
Below

CSS Improvements
7.2: Only display on home page the actual number of apps that will fit.
Remove more inline styles
Boxed display setting - limit home page to always being 4 at most

Squidly271 committed:
Release update to beta

Squidly271 committed:
Possible PHP error in certain circumstances

Unable to replicate

Squidly271 committed:
Release

Squidly271 committed:
Fix search bar not appearing 7.1 if display boxed

Squidly271 committed:
Test release

Squidly271 committed:
Fixed: Popup modals were allowing ESC to close

Should only allow ESC if outside click is also allowed

Squidly271 committed:
update

Squidly271 committed:
Update community.applications.plg

Squidly271 committed:
Fix searchbox / repository when showing specials

Squidly271 committed:
Merge branch 'css' of https://github.com/unraid/community.applications into css

Squidly271 committed:
Update community.applications-2025.07.22-x86_64-1.txz
